x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/autoscaling/MlAutoscalingResourceTracker.java

```diff
@@ -208,47 +218,98 @@ static void getMemoryAndProcessors(
 
             if (AssignmentState.STARTING.equals(assignment.getAssignmentState()) && assignment.getNodeRoutingTable().isEmpty()) {
-
+                // this state occurs when the assignment has just been added and has not been assigned to any nodes
                 logger.debug(
                     () -> format(
                         "trained model [%s] lacks assignment , memory required [%d]",
                         modelAssignment.getKey(),
                         estimatedMemoryUsage
                     )
                 );
-
-                extraSingleNodeModelMemoryInBytes = Math.max(extraSingleNodeModelMemoryInBytes, estimatedMemoryUsage);
+                extraPerNodeModelMemoryBytes = Math.max(extraPerNodeModelMemoryBytes, estimatedMemoryUsage);
                 extraModelMemoryInBytes += estimatedMemoryUsage;
 
-                // if not low priority, check processor requirements
+                // if not low priority, check processor requirements.
                 if (Priority.LOW.equals(modelAssignment.getValue().getTaskParams().getPriority()) == false) {
-                    // as assignments can be placed on different nodes, we only need numberOfThreadsPerAllocation here
-                    extraSingleNodeProcessors = Math.max(extraSingleNodeProcessors, numberOfThreadsPerAllocation);
-                    extraProcessors += numberOfAllocations * numberOfThreadsPerAllocation;
+                    if (numMissingProcessors > numberOfAvailableProcessors) {
+                        // as assignments can be placed on different nodes, we only need numberOfThreadsPerAllocation here
+                        extraSingleNodeProcessors = Math.max(extraSingleNodeProcessors, numberOfThreadsPerAllocation);
+                        extraProcessors += numMissingProcessors - numExistingProcessorsToBeUsed;
+                    }
+                    numberOfAvailableProcessors -= numExistingProcessorsToBeUsed;
                 }
             } else if (assignment.getNodeRoutingTable().values().stream().allMatch(r -> r.getState().consumesMemory() == false)) {
-                // Ignore states that don't consume memory, for example all allocations are failed
+                // Ignore states that don't consume memory, for example all allocations are failed or stopped
                 continue;
+            } else if (AssignmentState.STARTED.equals(assignment.getAssignmentState()) && numMissingAllocations > 0) {
+                // this state occurs when we have a partial allocation, but not fully allocated
+
+                long numMemoryBytesNeeded = assignment.getTaskParams().estimateMemoryUsageBytes();
+
+                if (numberOfAvailableProcessors < numMissingProcessors) {
+                    // we don't have enough processors or memory to start the new allocations, we need to scale up
+
+                    if (numberOfAvailableProcessors < numMissingProcessors) {
```

Comment:
This is the same conditional as on line 275, there's no need to repeat the check here.